### PR TITLE
test: refactor op e2e tests to use `BTCClient`

### LINKED ISSUE #468: op e2e: refactor bitcoin rpc url to use `BTCClient`
At the moment when running op e2e tests, we pass in a bitcoin mainnet rpc url to query the block finalization status.

We should refactor this to:
1. use the `BTCClient` interface as per [this](https://github.com/babylonchain/staking-indexer/blob/dev/btcclient/client.go#L17)
2. (optionally) spin up a local `bitcoind` note to bypass rpc rate limits

This issue is related to https://github.com/babylonchain/babylon-da-sdk/issues/41

### PR DESCRIPTION
**Pull request by parketh**
## Summary

At the moment when running op e2e tests, we pass in a Bitcoin mainnet rpc url to query the block finalization status.

This PR refactors tests to use the `BTCClient` interface in line with our update to the DA SDK, track in this issue:  https://github.com/babylonchain/babylon-da-sdk/issues/41

This PR closes #468.

## Test plan

```
make op-e2e-devnet
make test-e2e-op
```

### CODE REVIEW

**Review by SebastianElvis — COMMENTED** [review]
Code lgtm, CI failed though

**Comment by bap2pecs** [comment]
@SebastianElvis CI failed likely due to there are so many moving parts lol

I tested locally to glue everything up and it worked

**Comment by bap2pecs** [comment]
these are what is needed to make the CI pass:

- https://github.com/babylonchain/babylon-da-sdk/pull/45
- https://github.com/babylonchain/babylon-da-sdk/pull/39
- https://github.com/babylonchain/babylon-contract/pull/204
- https://github.com/babylonchain/babylon-da-sdk/pull/44

cc @SebastianElvis

**Comment by bap2pecs** [comment]
ok now fast sync is also fixed after all the changes. so I just enabled it in the e2e test

**Comment by bap2pecs** [comment]
once fast sync is enabled the test becomes flaky: https://gist.github.com/bap2pecs/f75f3134ad7d71865aa011d548c46056

so I disabled it for now. not going to fix in this PR

File: itest/opstackl2/op_test_manager.go [hunk 1]
```diff
@@ -176,11 +177,15 @@ func StartOpL2ConsumerManager(t *testing.T) *OpL2ConsumerTestManager {
 	err = fpApp.Start()
 	require.NoError(t, err)
 
-	// init SDK client
+	// 10. init SDK client
+	// We pass in an external Bitcoin RPC address but otherwise use the default configs.
+	// The RPC url must be trimmed to remove the http:// or https:// prefix.
+	btcConfig := btcclient.DefaultBTCConfig()
+	btcConfig.RPCHost = trimLeadingHttp(opSysCfg.DeployConfig.BabylonFinalityGadgetBitcoinRpc)
```
Comment (maurolacy) [inline itest/opstackl2/op_test_manager.go:184]: Nit: Can't you just use a host name instead of a url in the config?

**Review by maurolacy — APPROVED** [review]
Approving to unblock.

CI is not passing.

File: itest/opstackl2/op_test_manager.go (hunk 1 above)
Comment (bap2pecs) [reply itest/opstackl2/op_test_manager.go:184]: let us think about that. cc @parketh what's your thought?

**Comment by bap2pecs** [comment]
ok all CI passed

**Pull request merged**